pkg/controller/gitopsservice/gitopsservice_controller.go

```diff
@@ -47,8 +50,22 @@ var (
 	serviceNamespace                  = "openshift-gitops"
 	depracatedServiceNamespace        = "openshift-pipelines-app-delivery"
 	clusterVersionName                = "version"
+	argoClientID                      = "openshift-gitops"
+	keycloakRouteName                 = "keycloak"
+	keycloakSecretName                = fmt.Sprintf("keycloak-client-secret-%s", rhsso.KeycloakArgoClient)
+	argoCDSecretName                  = "argocd-secret"
 )
 
+const gitopsIdentifier = "openshift-gitops"
```

Comment:
Please seperate the OIDC/SSO stuff out from gitops service controller and create a new controller.

Reply:
Right.
Setting up of SSO needs to happen within the `GitOpsService` reconcile loop based on the `watch` on `GitOpsService`

Reply:
@sbose78 Why?  Because sso is watching the ArgoCD route?   GitOpsService is very clustered  (with the argocd stuff)  The sso stuff does not seem to belong to GitOpsService.

Reply:
The sso service gets created if the GitOpsService CR has `.spec.sso=true`. In that case, aren't we watching the `GitOpsService` CR ?

Reply:
By default, GitopsService CR has `.spec.sso=true`. gitopsservice_controller has a watch to `GitOpsService CR`

Reply:
@sbose78  I think we should create a new resource.  GitOpsService was created for the purpose of managing backend service which does not have much to do with Argo CD SSO.   Also, it is an opportunity to start using a correct api group if we create a new resource.